tools/helixer/helixer.xml

```diff
@@ -11,51 +11,59 @@
 
     <command detect_errors="exit_code"><![CDATA[
 
-        /usr/local/bin/fetch_helixer_models.py &&
         Helixer.py
         --fasta-path '$input'
         --species '$species'
-        --lineage $lineage.lineages
+        --lineage $lineages
         --gff-output-path '$output'
 
         --temporary-dir ./
 
-        --subsequence-length $lineage.subsequence_length
-        #if str($lineage.option_overlap.use_overlap) == "true":
-            --overlap-offset $lineage.option_overlap.overlap_offset
-            --overlap-core-length $lineage.option_overlap.overlap_core_length
+        #set default_subsequence_length = {"fungi": 21384, "land_plant": 106920, "invertebrate": 213840, "vertebrate": 213840}
+        #set subsequence_len = default_subsequence_length[$lineages]
+        #set default_overlap_offset = {"fungi": 10692, "land_plant": 53460, "invertebrate": 106920, "vertebrate": 106920}
+        #set overlap_off = default_overlap_offset[$lineages]
+        #set default_overlap_core_length = {"fungi": 16038, "land_plant": 80190, "invertebrate": 160380, "vertebrate": 160380}
+        #set overlap_core_len = default_overlap_core_length[$lineages]
+
+        #if str($subsequence_length) == "":
+            --subsequence-length '$subsequence_len'
         #else:
-            --no-overlap
+            --subsequence-length '$subsequence_length'
         #end if
+
+        #if str($option_overlap.use_overlap) == "true":
+            #if str($option_overlap.overlap_offset) == "":
+                --overlap-offset '$overlap_off'
+            #else:
+                --overlap-offset '$overlap_offset'
+            #end if
+
+            #if str($option_overlap.overlap_core_length) == "":
+                --overlap-core-length '$overlap_core_len'
+            #else:
+                --overlap-core-length '$overlap_core_length'
+            #end if
+        #end if
+
```

Comment:
I want to ensure that if the subsequence-length, overlap-offset and overlap-core-length parameters are empty, then the default value defined in the lineage-based dictionaries is selected. I've added [$lineages].
Here's the error I got 

![image](https://github.com/genouest/galaxy-tools/assets/127295521/3cf6ce4d-71cc-4ee3-ab34-e3af380f779e)

Reply:
oh, have you tried something like `#set subsequence_len = default_subsequence_length.get($lineages)`  maybe?

Reply:
I tried it. 
Here's the error message : 

![image](https://github.com/genouest/galaxy-tools/assets/127295521/8f19f5a0-18df-41a9-bc8a-4d549bdbcf1a)

Reply:
ah `#set subsequence_len = default_subsequence_length.get(str($lineages))` maybe? or `#set subsequence_len = default_subsequence_length.[str($lineages)]`?